src/hooks/toast/useLoadingToast.tsx

```diff
@@ -0,0 +1,78 @@
+import { ReactNode } from "react"
+import toast from "react-hot-toast"
+
+import { ErrorToast, LoadingToast, SuccessToast } from "@/components/Toast"
+
+export type ToastProps = {
+  position:
+    | "top-left"
+    | "top-center"
+    | "top-right"
+    | "bottom-left"
+    | "bottom-center"
+    | "bottom-right"
+  duration: number
+}
+
+type LoadingToastProps = ToastProps & {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  onSuccess: (t: any) => ReactNode
+  onError: (e: Error) => ReactNode
+}
+
+export function useLoadingToast<T>(
+  promise: Promise<T>,
```

Comment:
I think this is probably from the `react-hot-toast` documentation, but in our case its a bit different because we are waiting for a transaction to confirm and we do not have a promise. 

What I would do is:
- pass `txHash` to <LoadingToast />
- wait for the transaction in `LoadingToast` with `useWaitForTransaction` + the passed `txHash`
- in `onSettled` dismiss the toast

We can improve upon that by handling error/success within the component as well, but I think it would be better to focus on making the baseline toasts as clean as possible first. Can be iterated upon.

Reply:
I get your point. The LoadingToast becomes quite specialized for the transaction waiting as a result, don't you think? I considered taking a more open-ended approach and allowing any method that requires LoadingToast to utilize it, rather than only for transaction waiting.

Reply:
I included my reply to this in my reply to the previous comment. I think it's best to just be specific. This is our primary use case for loading toasts. Build around it.